Marlin/src/HAL/STM32/inc/SanityCheck.h

```diff
@@ -47,9 +47,9 @@
 #endif
 
 #if ENABLED(SERIAL_STATS_MAX_RX_QUEUED)
-  #error "SERIAL_STATS_MAX_RX_QUEUED is not supported on this platform."
+  #error "SERIAL_STATS_MAX_RX_QUEUED is not supported on STM32."
 #elif ENABLED(SERIAL_STATS_DROPPED_RX)
-  #error "SERIAL_STATS_DROPPED_RX is not supported on this platform."
+  #error "SERIAL_STATS_DROPPED_RX is not supported on STM32."
```

Comment:
First: Excellent work and thanks for the update!
While you've fixed this, I was thinking that my changes could work almost unchanged on all STM32**F** series (STM32F1, STM32F2, STM32F4) since the register map is compatible as far as I've read. That would require dealing with registers directly (bypassing stm32duino/libmaple abstraction layer) but it's almost done already (but commented). I'm not sure it'll work on STM32M series (I haven't checked). So, maybe we should split the STM32 HAL's folder to 2 of them, STM32F and STM32M. 
Is there any board with a STM32M series ?

Reply:
We are currently trying to eliminate extra STM32 HALs, not add more 🙂

We recently deleted the STM32F4 HAL. STM32F1 will be deleted soon enough.

By STM32M series I assume you mean STM32MP1? These are mixed-core processors which could allow running Linux on an A7 core and Marlin on an M4 core. I don't think we need to worry about those for now.

Reply:
Great! When doing this cleanup, I'll replace all libmaple specific code with plain registers access that'll work on all STM32F versions.